src/base/static/components/story-sidebar/index.js

```diff
@@ -0,0 +1,194 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { Map, OrderedMap, fromJS } from "immutable";
+import Spinner from "react-spinner";
+
+import StoryChapter from "../molecules/story-chapter";
+import constants from "../../constants";
+
+import { translate } from "react-i18next";
+
+import { getModelFromUrl } from "../../utils/collection-utils";
+
+import "./index.scss";
+
+import { story as storyConfig, place as placeConfig } from "config";
+
+class StorySidebar extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      currentStoryHeader: "",
+      currentStoryDescription: "",
+      currentStoryName: "",
+      currentStoryChapters: OrderedMap(),
+      currentStoryRoute: "",
+      currentChapter: Map(),
+      isStoryDataReady: false,
+      isWithError: false,
+    };
+  }
+
+  componentDidMount() {
+    Promise.all(this.props.placeCollectionPromises)
+      .then(() => {
+        this.stories = Object.keys(storyConfig).reduce((stories, storyName) => {
+          return stories.set(
+            storyName,
+            Object.keys(storyConfig[storyName].order).reduce(
+              (chapters, chapterUrl) => {
+                return chapters.set(
+                  chapterUrl,
+                  fromJS(
+                    getModelFromUrl(this.props.places, chapterUrl).attributes,
+                  ),
+                );
+              },
+              OrderedMap(),
+            ),
+          );
+        }, OrderedMap());
+
+        const currentStoryState = this.getCurrentStoryState(
+          Backbone.history.getFragment().split("/"),
+          false,
+        );
+
+        currentStoryState.isStoryDataReady = true;
+        this.setState(currentStoryState);
+
+        this.props.router.on("route", (fn, route) => {
+          this.setState(this.getCurrentStoryState(route));
+        });
+      })
+      .catch(() => {
+        console.log("ERROR!");
+        this.setState({
+          isWithError: true,
+        });
+      });
+  }
+
+  getCurrentStoryState(route, isInitialized = true) {
+    const joinedRoute = route.join("/");
+    const [foundStoryName, foundStory] =
+      this.stories.findEntry(story => {
+        return story.get(joinedRoute);
+      }) || [];
+
+    if (foundStory) {
+      return {
+        currentStoryHeader: storyConfig[foundStoryName].header,
+        currentStoryDescription: storyConfig[foundStoryName].description,
+        currentStoryName: foundStoryName,
+        currentStoryChapters: foundStory,
+        currentStoryRoute: joinedRoute,
+        currentChapter: foundStory.get(joinedRoute),
```

Comment:
I think this disparate state, along with the `getIconUrl` and `getTitle` methods, is a sign that we can benefit from a "story" abstraction. What do you think?

I imagine that we can have a "storyFactory" function that will accept all relevant state and return a POJO representation of the story. We can then treat that POJO as a first class representation of our "story", which will be the hydrated representation of the "story" we have saved in our config. That would abstract away all story-state-related implementation details, which seem to be cluttering this component.

Probably out of scope of this PR, but let me know what you think.

Reply:
I see what you're getting at. I do think a story abstraction would be useful, and ultimately I think we should have a notion of a story abstraction on the backend, especially as we move toward executing a GUI story builder.

If we make a story factory, where do you think it would live inside the current organization of the app? Would it be another `util`?

Reply:
I think another `util` is fine, or a `src/base/models/story.js` or `src/base/state/story.js` file could work as well. Eventually, we will be adding a `state` dir when we add redux to our state management, and the logic will live there.